zerver/views/webhooks/trello/__init__.py

```diff
@@ -0,0 +1,30 @@
+# Webhooks for external integrations.
+from __future__ import absolute_import
+import ujson
+from zerver.models import get_client
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+
+from .card_actions import SUPPORTED_CARD_ACTIONS, process_card_action
+from .exceptions import UnsupportedAction
+
+
+@api_key_only_webhook_view
+@has_request_variables
+def api_trello_webhook(request, user_profile, stream=REQ(default='trello')):
+    payload = ujson.loads(request.body)
+    action_type = payload.get('action').get('type')
+    try:
+        subject, body = get_subject_and_body(payload, action_type)
+    except UnsupportedAction:
+        return json_error('Unsupported action_type: {action_type}'.format(action_type=action_type))
+
+    check_send_message(user_profile, get_client('ZulipTrelloWebhook'), 'stream', [stream], subject, body)
+    return json_success()
+
+
+def get_subject_and_body(payload, action_type):
+    if action_type in SUPPORTED_CARD_ACTIONS:
+        return process_card_action(payload, action_type)
+    raise UnsupportedAction('{} if not supported'.format(action_type))
```

Comment:
Ahh, I see, because you're imagining having more action types.  OK, that's reasonable.